destination.go

```diff
@@ -26,7 +26,7 @@ type DestinationConfig struct {
 
 func NewDestination() sdk.Destination {
 	// Create Destination and wrap it in the default middleware.
-	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+	return sdk.DestinationWithMiddleware(new(Destination), sdk.DefaultDestinationMiddleware()...)
```

Comment:
Why making this change?

Reply:
I tried the exhaustruct linter, which is one that I usually use in my personal projects. But it is a bit too much of a radical code style change, so I removed it.

This is a leftover of that implementation, I'll remove it for consistency.